File: server/tests/fixtures/ast_chunker/sample.ts
// Fixture for ast_chunker TS tests.

export interface User {
    id: number;
    name: string;
}

export function makeUser(id: number, name: string): User {
    return { id, name };
}

const upper = (s: string): string => {
    return s.toUpperCase();
};

export class UserService {
    private cache: Map<number, User> = new Map();

    public getUser(id: number): User | undefined {
        return this.cache.get(id);
    }
}

File: server/tests/fixtures/ast_chunker/sample.py
"""Fixture for ast_chunker Python tests."""

import math


def add(a, b):
    return a + b


async def fetch_user(user_id):
    return {"id": user_id, "name": "test"}


class Calculator:
    """A simple calculator."""

    def __init__(self, initial=0):
        self.value = initial

    def add(self, other):
        self.value += other
        return self.value

    def multiply(self, other):
        self.value *= other
        return self.value


def _internal_helper():
    return math.pi
